pyvisa/resources/messagebased.py

```diff
@@ -469,17 +469,44 @@ def read_binary_values(self, datatype='f', is_big_endian=False,
         :rtype: type(container)
 
         """
-        block = self._read_raw()
+        block = b''
         expected_length = 0
 
         if header_fmt == 'ieee':
-            offset, data_length = util.parse_ieee_block_header(block)
+            hash = self.read_bytes(1)                   # Hash '#'
+            digits = self.read_bytes(1)                 # Digits
+            block_size = self.read_bytes(int(digits))   # Block size
```

Comment:
We need to test for digits being 0 here. The Keithley 2000 for example return such a block which is not technically ieee compliant but is close enough. To allow that kind of use case, we could add an `expected_length` keyword argument to `read_binary_values` and `query_binary_values`, that would default to `None`, allowing to write: `expected_length = expected_length or int(block_size)`.

Reply:
The case for 0 has been tested. If it is zero it will read 0 bytes so remain the same.